pkgs/build-support/rust/build-rust-crate/build-crate.nix

```diff
@@ -13,6 +13,17 @@
             (if release then "-C opt-level=3" else "-C debuginfo=2")
             (["-C codegen-units=1"] ++ extraRustcOpts);
         rustcMeta = "-C metadata=${metadata} -C extra-filename=-${metadata}";
+
+    # Some platforms have different names for rustc.
+    rustPlatform =
+      with stdenv.hostPlatform.parsed;
+      let cpu_ = if cpu.name == "armv7a" then "armv7"
+                 else cpu.name;
```

Comment:
I'd rather have a fixed and known-to-work name mapping instead of such hardcoded special cases.

Reply:
Me too!
If you know of any such naming, great. Rustc doesn't seem to respect GNU conventions, NixOS does. I don't want to sound scary, but most, if not all, of `rustBuildCrate` was built by trial and error.

Reply:
Asking in #rustc they pointed me to https://github.com/rust-lang/rust/tree/master/src/librustc_target/spec where this stuff is apparently declared, soo if somebody feels inclined to do a mapping for them there you go :P

Reply:
They also said they match llvm targets, and those are defined in each file as well

Reply:
You can get list of supported target names by running `rustc --print target-list`.